File: Software Engineering Artifact/Task.py
# -*- coding: utf-8 -*-
"""
Created on Tue Jan 11 13:33:02 2022

@author: danielle parham
Class: CS 499
Category: Software Design and Engineering
title: Task 

"""
#task_func class
#designed to limit task variables: taskID, taskName, and taskDes
#to prevent invalid input: char limit or None values
class task_func:
    def task_details(self, taskID, taskName, taskDes):
        
        #verify taskID is valid
        if taskID == None or len(taskID) > 10:
            #calls error message statement
            self.errMessage()
        
        #verify taskName is valid
        if taskName == None or len(taskName) >20:
            self.errMessage()
        
        #verify task Description is valid
        if taskDes == None or len(taskDes)>50:
            self.errMessage()
            
        #default else statement
        else:
        
            print("Confirmed: Task Added")
            self._taskID = taskID
            self._taskName = taskName
            self._taskDes = taskDes
            
    # setters and getters
    
    def getTaskID(self):
       
       return self._taskID
        
    def setTaskID(self, taskID):
        self._taskID = taskID
    
    def getTaskName(self):
        return self._taskName
    
    def setTaskName(self, taskName):
        self._taskName = taskName
    
    def getTaskDes(self):
        return self._taskDes
    
    def setTaskDes(self, taskDes):
        self._taskDes = taskDes
    
    #error message to be printed if conditions are met
    def errMessage(self):
        raise Exception("Invalid Input")            
        

File: Software Engineering Artifact/Task.java
package module4;

public class Task {
	private static String taskID;
	private static String taskName;
	private static String taskDes;
	public Object getTaskID;
	

	public Task(String taskID, String taskName, String taskDes) {
		if(taskID == null || taskID.length()>10) {
			throw new IllegalArgumentException("Invalid Task ID");
			
		}
		if(taskName == null || taskName.length()>20) {
			throw new IllegalArgumentException("Invalid Task");
		}
		if(taskDes == null || taskDes.length()>50) {
			throw new IllegalArgumentException("Invalid Task Description");
		}
		this.taskID = taskID;
		this.taskName = taskName;
		this.taskDes = taskDes;
		
	}

	public String getTaskID() {
		return taskID;
	}

	public void setTaskID(String taskID) {
		Task.taskID = taskID;
	}

	public String getTaskName() {
		return taskName;
	}

	public void setTaskName(String taskName) {
		this.taskName = taskName;
	}

	public String getTaskDes() {
		return taskDes;
	}

	public void setTaskDes(String taskDes) {
		this.taskDes = taskDes;
	}
//	public String getUniqueID() {
	//	return uniqueID;
//	}

	//public String setUniqueID(String uniqueID) {
//		return this.uniqueID = uniqueID;
//	}

	
}
